core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceConfigurator.java

```diff
@@ -0,0 +1,164 @@
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import com.linecorp.armeria.internal.annotation.AnnotatedHttpService;
+import com.linecorp.armeria.server.annotation.ExceptionHandlerFunction;
+import com.linecorp.armeria.server.annotation.RequestConverterFunction;
+import com.linecorp.armeria.server.annotation.ResponseConverterFunction;
+
+/**
+ * A configurator to configure {@link ExceptionHandlerFunction}s, {@link RequestConverterFunction}s
+ * or {@link ResponseConverterFunction}s for an {@link AnnotatedHttpService}.
+ */
+final class AnnotatedHttpServiceConfigurator {
+
+    /**
+     * Creates a new instance with the specified {@code exceptionHandlersAndConverters}.
+     */
+    static AnnotatedHttpServiceConfigurator ofExceptionHandlersAndConverters(
+            Iterable<?> exceptionHandlersAndConverters) {
+
+        Builder<ExceptionHandlerFunction> exceptionHandlers = null;
+        Builder<RequestConverterFunction> requestConverters = null;
+        Builder<ResponseConverterFunction> responseConverters = null;
+
+        for (final Object o : exceptionHandlersAndConverters) {
+            if (o instanceof ExceptionHandlerFunction) {
+                if (exceptionHandlers == null) {
+                    exceptionHandlers = ImmutableList.builder();
+                }
+                exceptionHandlers.add((ExceptionHandlerFunction) o);
+            } else if (o instanceof RequestConverterFunction) {
+                if (requestConverters == null) {
+                    requestConverters = ImmutableList.builder();
+                }
+                requestConverters.add((RequestConverterFunction) o);
+            } else if (o instanceof ResponseConverterFunction) {
+                if (responseConverters == null) {
+                    responseConverters = ImmutableList.builder();
+                }
+                responseConverters.add((ResponseConverterFunction) o);
+            } else {
+                throw new IllegalArgumentException(o.getClass().getName() +
+                                                   " is neither an exception handler nor a converter.");
+            }
+        }
+
+        final List<ExceptionHandlerFunction> exceptionHandlerFunctions =
+                exceptionHandlers != null ? exceptionHandlers.build() : ImmutableList.of();
+        final List<RequestConverterFunction> requestConverterFunctions =
+                requestConverters != null ? requestConverters.build() : ImmutableList.of();
+        final List<ResponseConverterFunction> responseConverterFunctions =
+                responseConverters != null ? responseConverters.build() : ImmutableList.of();
+
+        return ofExceptionHandlersAndConverters(exceptionHandlerFunctions, requestConverterFunctions,
```

Comment:
Let's do not call this method and just create the instance here so that we can
1 - remove all setters
2 - make the private fields final

Reply:
I meant doing `new AnnotatedHttpServiceConfigurator(exceptionHandlers.build(), requestConverters.build(), responseConverters.build())`.
Sorry for making you get confused. 😉

Reply:
Oh... it's fine. I should have asked again 
I'll edit the details of the review later. :smile: